device/cuda/include/cuda/seeding/track_params_estimation.hpp

```diff
@@ -0,0 +1,41 @@
+/** TRACCC library, part of the ACTS project (R&D line)
+ *
+ * (c) 2021 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#pragma once
+
+#include <cuda/seeding/track_params_estimating.cuh>
+
+namespace traccc {
+namespace cuda {
+
+/// Seed finding for cuda
+class track_params_estimation {
+    public:
+    using input_type =
+        std::pair<host_measurement_container&, host_seed_container&>;
+    using output_type = host_bound_track_parameters_collection;
+
+    track_params_estimation(vecmem::memory_resource* mr)
+        : m_mr(mr), params(mr) {}
+
+    output_type& operator()(input_type& i) {
```

Comment:
Again, returning a non-const reference to a member object seems terribly thread-unsafe. I would've just opted for copy elision on return in this function...

Reply:
answered above